pkg/tfbridge/provider.go

```diff
@@ -1158,18 +1158,43 @@ func (p *Provider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*pulum
 		return nil, errors.Wrapf(err, "diffing %s", urn)
 	}
 
-	dd := makeDetailedDiffExtra(ctx, schema, fields, olds, news, diff)
-	detailedDiff, changes := dd.diffs, dd.changes
+	var detailedDiff map[string]*pulumirpc.PropertyDiff
+	var changes pulumirpc.DiffResponse_DiffChanges
 
-	if opts.enableAccurateBridgePreview {
-		if decision := diff.DiffEqualDecisionOverride(); decision != shim.DiffNoOverride {
-			if decision == shim.DiffOverrideNoUpdate {
-				changes = pulumirpc.DiffResponse_DIFF_NONE
-			} else {
-				changes = pulumirpc.DiffResponse_DIFF_SOME
-			}
+	decisionOverride := diff.DiffEqualDecisionOverride()
+	if opts.enableAccurateBridgePreview && decisionOverride != shim.DiffNoOverride {
+		if decisionOverride == shim.DiffOverrideNoUpdate {
+			changes = pulumirpc.DiffResponse_DIFF_NONE
+		} else {
+			changes = pulumirpc.DiffResponse_DIFF_SOME
 		}
+
+		// We need to compare the new and olds after all transformations have been applied.
+		// ex. state upgrades, implementation-specific normalizations etc.
+		proposedState, err := diff.ProposedState(res.TF, state)
```

Comment:
Nit: wouldn't mind if this prep code went into detaield_diff.go file to keep provider.go higher level. It makes total sense to have it though! Somewhere. Thanks!